src/Http/Routing/src/Builder/RouteHandlerFilterExtensions.cs

```diff
@@ -56,27 +57,77 @@ public static RouteHandlerBuilder AddFilter(this RouteHandlerBuilder builder, IR
         return builder;
     }
 
+    /// <summary>
+    /// Registers a filter of type <typeparamref name="TFilterType"/> onto the route handler.
+    /// </summary>
+    /// <typeparam name="TFilterType">The type of the <see cref="IRouteHandlerFilter"/> to register.</typeparam>
+    /// <param name="builder">The <see cref="RouteHandlerBuilder"/>.</param>
+    /// <returns>A <see cref="RouteHandlerBuilder"/> that can be used to further customize the route handler.</returns>
+    public static RouteHandlerBuilder AddRouteHandlerFilter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TFilterType>(this RouteHandlerBuilder builder)
+        where TFilterType : IRouteHandlerFilter
+    {
+        // We have a RouteHandlerBuiler and GroupRouteBuilder-specific AddFilter methods for convenience so you don't have to specify both arguments most the time.
+        return builder.AddRouteHandlerFilter<RouteHandlerBuilder, TFilterType>();
+    }
+
+    /// <summary>
+    /// Registers a filter of type <typeparamref name="TFilterType"/> onto the route handler.
+    /// </summary>
+    /// <typeparam name="TFilterType">The type of the <see cref="IRouteHandlerFilter"/> to register.</typeparam>
+    /// <param name="builder">The <see cref="RouteHandlerBuilder"/>.</param>
+    /// <returns>A <see cref="RouteHandlerBuilder"/> that can be used to further customize the route handler.</returns>
+    public static RouteGroupBuilder AddRouteHandlerFilter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TFilterType>(this RouteGroupBuilder builder)
+        where TFilterType : IRouteHandlerFilter
+    {
+        // We have a RouteHandlerBuiler and GroupRouteBuilder-specific AddFilter methods for convenience so you don't have to specify both arguments most the time.
+        return builder.AddRouteHandlerFilter<RouteGroupBuilder, TFilterType>();
+    }
+
     /// <summary>
     /// Registers a filter given a delegate onto the route handler.
     /// </summary>
     /// <param name="builder">The <see cref="RouteHandlerBuilder"/>.</param>
-    /// <param name="routeHandlerFilter">A <see cref="Delegate"/> representing the core logic of the filter.</param>
+    /// <param name="routeHandlerFilter">A <see cref="Func{T1, T2, TResult}"/> representing the core logic of the filter.</param>
```

Comment:
Not a fan of either `Delegate` or `Func<T1, T2, TResult>`. Maybe just say "A method representing ..."